packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts

```diff
@@ -112,7 +112,25 @@ function patchSdk(awsSdk: any): any {
 }
 
 /* eslint-disable @typescript-eslint/no-require-imports, import/no-extraneous-dependencies */
-export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent, context: AWSLambda.Context) {
+export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent, _context: AWSLambda.Context) {
+  console.log(JSON.stringify(event));
+  event.ResourceProperties.Create = decodeCall(event.ResourceProperties.Create);
+  event.ResourceProperties.Update = decodeCall(event.ResourceProperties.Update);
+  event.ResourceProperties.Delete = decodeCall(event.ResourceProperties.Delete);
```

Comment:
If any of these fail the same error can still happen. The only safe initialization I think is:

```
let physicalResourceId = event.PhysicalResourceId ?? context.logStreamName;
```

And if we do that, all the other code changes aren't necessary anymore, correct?

Reply:
Thanks @rix0rrr - the PhysicalResourceID isn't accessible for create events so generates the following error:
`Property 'physicalResourceId' does not exist on type 'CloudFormationCustomResourceCreateEvent'`

I've updated it to default to context.logStreamName and update as it was before in the pipeline, let me know if there's issues with this approach.

Have we got any existing tests that need updating? Or do we have any examples of how to go about testing the runtime lambda?

Reply:
I see. In order to properly compile here, we would need a type test.

You should be able to do this:

```ts
let physicalResourceId = event.RequestType === 'Create' ? context.logStreamName : event.PhysicalResourceId;
```